app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,35 @@
+class OrdersController < ApplicationController
+  load_and_authorize_resource
+
+  def index
+    @menus = Menu.all
+    @orders = params[:menu_id] ? Order.for_menu(params[:menu_id]) : Order.all
+  end
+
+  def new
+    @menu = Menu.find(params[:menu_id])
+  end
+
+  def create
+    menu_item_ids = MenuItem::TYPES_OF_DISHES.map {|dish_type| params["#{dish_type.to_s}_id"] }.compact
+    redirect_to menus_path and return if menu_item_ids.empty?
+
+    menu_items = MenuItem.find(menu_item_ids)
+    menu = Menu.find(order_params[:menu_id])
+    total_price = menu_items.sum(&:price)
+    order = Order.new(user_id: current_user.id, total_price: total_price, menu_id: menu.id)
```

Comment:
this is overall quite messy method. Split it into separate methods, make a single publit entry method for all of this.
Also, I'd say you are depending on params[:menu_id] a little too much. would be better to initialize @menu in before_action based on this param and manipulate with object

Reply:
@volodialeobit 
Thanks for comment. I agree with your point, but not really understood why method must be public and I guess it is extra job to split it into separate methods, it will be clear I guess if I call this method "generate_new_order_with_relations", or maybe I didn't understand you right?

Reply:
https://medium.com/@scottdomes/service-objects-in-rails-75ca74214b77
pay extra attention to chapter "Syntactic sugar" - that's the way we usually go with. 
long story short: you should have a SINGLE PUBLIC method in service (it's better to have a common name all around the project - for example `call`, while the meaning of this service should be presented in the name of service), and private methods inside the service

Reply:
Ah right, I thought you mean one more public method despite of call method )